types/react-side-effect/react-side-effect-tests.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+import withSideEffect = require("react-side-effect");
+
+interface DocumentTitleProps {
+    title: string;
+}
+
+type State = string | undefined;
+
+interface ServerState {
+    innerState: State;
+}
+
+const DocumentTitle = () => null;
+
+function reducePropsToState(propsList: DocumentTitleProps[]): State {
+    const innermostProps = propsList[propsList.length - 1];
+    if (innermostProps) {
+        return innermostProps.title;
+    }
+}
+
+function handleStateChangeOnClient(title: State) {
+    document.title = title || "";
+}
+
+const mapStateOnServer = (state: State): ServerState => ({ innerState: state });
+
+const DocumentTitleServer = withSideEffect(
+    reducePropsToState,
+    handleStateChangeOnClient,
+    mapStateOnServer
+)(DocumentTitle);
+
+const testServer = () => {
+    const testComponent = () => <DocumentTitleServer title="Title" />;
+    const peekedState: ServerState = DocumentTitleServer.peek();
+    const rewindedState: State = DocumentTitleServer.rewind();
```

Comment:
This should be:

```js
const peekedState: State | ServerState = DocumentTitleServer.peek();
const rewindedState: ServerState = DocumentTitleServer.rewind();
```

Reply:
See above. You're right about rewind, wrong about peek.